### PR TITLE
Fixing most of the warnings when compiling

### PR DESCRIPTION
**Pull request by khanghugo**
Some warnings seem really tricky to fix because they are auto-generated by bootstrap.sh, which will call some other tools. Those warnings include weird 
```
warning: AC_LANG_CONFTEST: no AC_LANG_SOURCE call detected in body
```

or 
```
ar: `u' modifier ignored since `D' is the default (see `U')
```

The latter is minimized.

Some of the changes here are also to please the compiler. Eg: memmove instead of memcopy for overlapped memory region. Not sure about the changes in its behavior but it should be fine.

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -5,13 +5,13 @@ dnl Licensed under LGPL. See file COPYING.
 dnl
 dnl ref: http://www.gnu.org/software/autoconf/manual/autoconf.info.gz
 dnl
-AC_PREREQ(2.57)
+AC_PREREQ([2.71])
```
Comment (davehorton) [inline configure.ac:8]: I dont want this requirement.  I thought you were going to revert these changes to configure.ac?

**Review by davehorton — CHANGES_REQUESTED** [review]

**Comment by davehorton** [comment]
With the latest change I am now dealing with this
```
make[3]: Nothing to be done for `built-sources'.
Making all in su
/Applications/Xcode.app/Contents/Developer/usr/bin/make  all-am
	 LTCOMPILE su.lo
../../libtool: line 1284: func_opt_split: command not found
libtool: Version mismatch error.  This is libtool 2.2.6b, but the
libtool: definition of this LT_INIT comes from libtool 2.4.6.
libtool: You should recreate aclocal.m4 with macros from libtool 2.2.6b
libtool: and run autoconf again.
make[4]: *** [su.lo] Error 1
make[3]: *** [all] Error 2
make[2]: *** [all-recursive] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
```

I am not sure you understood me.  I want *all* the changes you made to configure.ac reverted.

**Comment by khanghugo** [comment]
Hey, I did not think you would jump on it that quick.. Yea I did revert the AC_PREREQ after some inspections.
Let me revert the entire configure.ac, without running any autoupdate, and see if I can resolve the deprecated macros.

**Comment by khanghugo** [comment]
Hmmm, even before or after reverting configure.ac, it all compiles just fine for me. Maybe the newer version just works. Right now, I am in the process of installing the libtool version you are having and see if I can work on that.

Meanwhile, you can try the latest commit, which totally reverts all of my changes in configure.ac.

**Comment by khanghugo** [comment]
I have tested with libtool-2.2.6b and it works just fine. I think your error earlier is just me not really reverting everything.

**Comment by khanghugo** [comment]
Moreover, if there is openssl warning then I can change that, definitely. It is just that the original openssl used is v1.x.x
```
ctx = SSL_CTX_new(TLSv1_client_method());
```

So the warning will ask to specify the newer version of openssl in the system.

**Comment by davehorton** [comment]
I am not worried about the openssl warnings, those are ok for now.

I am still seeing errors in sip_pref_util.c though
```
sip_pref_util.c:328:12: warning: adding 'int' to a string does not append to the string [-Wstring-plus-int]
    base = MATCH("audio") || MATCH("automata") || MATCH("application") ||
           ^~~~~~~~~~~~~~
sip_pref_util.c:315:31: note: expanded from macro 'MATCH'
  (su_casenmatch(param + 1, s + 1, strlen(s) - 1) && \
                            ~~^~~
sip_pref_util.c:328:12: note: use array indexing to silence this warning
sip_pref_util.c:315:31: note: expanded from macro 'MATCH'
  (su_casenmatch(param + 1, s + 1, strlen(s) - 1) && \
                              ^
sip_pref_util.c:328:30: warning: adding 'int' to a string does not append to the string [-Wstring-plus-int]
    base = MATCH("audio") || MATCH("automata") || MATCH("application") ||
                             ^~~~~~~~~~~~~~~~~
sip_pref_util.c:315:31: note: expanded from macro 'MATCH'
  (su_casenmatch(param + 1, s + 1, strlen(s) - 1) && \
                            ~~^~~
sip_pref_util.c:328:30: note: use array indexing to silence this warning
sip_pref_util.c:315:31: note: expanded from macro 'MATCH'
  (su_casenmatch(param + 1, s + 1, strlen(s) - 1) && \
```

**Comment by khanghugo** [comment]
Okay, I see you are using clang while I am using gcc. Let me switch to that and see....

**Comment by khanghugo** [comment]
Okay, I think this might be it, beside from the obsolete warnings from ./configure and the ar cru flags complaints.

**Comment by davehorton** [comment]
can you rebase to the latest main branch?

**Comment by khanghugo** [comment]
Sure, meanwhile, I will revert more changes that gcc complains that clang did not.

**Comment by khanghugo** [comment]
Okay, I think I don't know what I am doing right now...

**Comment by davehorton** [comment]
haha, that is ok, hold for now.  This is good work, but I think what I may do is create my own branch and review and copy over some of these fixes.

**Comment by khanghugo** [comment]
Okay, I definitely don't know what I am doing.

**Comment by khanghugo** [comment]
Yep, everything definitely goes bad now. Please just ignore all of the later commit where I attempt to rebase.

**Comment by khanghugo** [comment]
The latest usable commit is https://github.com/davehorton/sofia-sip/pull/7/commits/4b98a7146a4c3dd7833eb07b6e35e714fd200d02

**Pull request closed without merging**